src/components/site-form.tsx

```diff
@@ -475,9 +487,15 @@ export const SiteForm = ( {
 											</div>
 										) }
 
-										{ setUseCustomDomain && setCustomDomain && (
-											<div className="text-a8c-gray-50 text-xs mt-2">
-												{ __( 'Your system password will be required to set up the domain.' ) }
+										{ useCustomDomain && setEnableSSL && (
+											<div className="flex items-center gap-2 mt-4">
+												<input
+													type="checkbox"
+													id="enable-ssl"
+													checked={ enableSSL }
+													onChange={ ( e ) => setEnableSSL( e.target.checked ) }
+												/>
+												<label htmlFor="enable-ssl">{ __( 'Enable SSL' ) }</label>
```

Comment:
```suggestion
												<label htmlFor="enable-ssl">{ __( 'Enable HTTPS' ) }</label>
```
HTTPS conveys the meaning more effectively and is technically more accurate.

Reply:
I had it as HTTP initially but I switched to be consistent with Local, users might already be familiar with that. WDYT?

Reply:
I think HTTPS conveys the meaning better here. We can consider this from an SEO perspective when writing from the blog and make sure to mention "SSL" in the blog post.